EventAuxReader.h

```diff
@@ -1,24 +1,68 @@
 #if !defined(EventAuxReader_h)
 #define EventAuxReader_h
 
+#include <cassert>
+
 #include "EventIdentifier.h"
+#include "art/EventAuxiliary.h"
 #include "cms/EventAuxiliary.h"
 
+#include "TFile.h"
+class TBranch;
+
 namespace cce::tf {
-class EventAuxReader {
-public:
-  EventAuxReader(void** iAddress): address_(iAddress){}
-
-  EventIdentifier doWork() {
-    if(address_) {
-      auto aux = *reinterpret_cast<edm::EventAuxiliary**>(address_); 
-      return EventIdentifier{aux->run(), aux->luminosityBlock(), aux->event()};
-    } else {
-      return EventIdentifier{0,0,0};
+
+  inline EventIdentifier cmsEventID(void** address) {
+    assert(address);
+    auto aux = *reinterpret_cast<edm::EventAuxiliary**>(address);
+    return EventIdentifier{aux->run(), aux->luminosityBlock(), aux->event()};
+  }
+
+  inline EventIdentifier artEventID(void** address) {
+    assert(address);
+    auto aux = *reinterpret_cast<art::EventAuxiliary**>(address);
+    return EventIdentifier{aux->run(), aux->subRun(), aux->event()};
+  }
+
+  inline std::function<EventIdentifier(void**)> select_reader(TFile& file)
+  {
+    if (file.GetKey("RootFileDB")) {
+      return artEventID;
     }
+    return cmsEventID;
   }
-private:
-  void** address_;
-};
+
+  class EventAuxReader {
+  public:
+    EventAuxReader() = default;
+
+    explicit EventAuxReader(TFile& file)
+      : func_{select_reader(file)}
+    {}
+
+    EventAuxReader& bindToBranch(void** branchToRead) noexcept
+    {
+      boundBranch_ = reinterpret_cast<void**>(branchToRead);
+      return *this;
+    }
+
+    EventAuxReader& bindToBranch(TBranch* branchToRead) noexcept
+    {
+      return bindToBranch(reinterpret_cast<void**>(branchToRead));
```

Comment:
This looks wrong. Shouldn't this be
```suggestion
      return bindToBranch(reinterpret_cast<void**>(branchToRead->GetAddress()));
```

Reply:
Oops.  You're right.  Will fix.